python/cudf/cudf/core/column/datetime.py

```diff
@@ -441,7 +464,11 @@ def _binaryop(self, other: ColumnBinaryOperand, op: str) -> ColumnBase:
         if out_dtype is None:
             return NotImplemented
 
-        return libcudf.binaryop.binaryop(lhs, rhs, op, out_dtype)
+        result_col = libcudf.binaryop.binaryop(lhs, rhs, op, out_dtype)
+        if out_dtype != cudf.dtype(np.bool_) and op == "__add__":
+            return result_col  # .astype(lhs.dtype)
```

Comment:
If `# .astype(lhs.dtype)` no longer needed, should the commented code be removed?